### PR TITLE
Run CI on main branch

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Run CI (verification and test deployment) on merge to `main` branch.

## Why it does that

Provide a baseline to verify that the CI on main branch is green.

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 name: CD
 
 on:
+  push:
```
Comment (cpeeyush) [inline .github/workflows/cd.yaml:4]: What about cloud-cd.yaml ? That should also be included to make sure that cloud deployment works fine on main(edc mvd upstream), correct ?

**Comment by cpeeyush** [comment]
@algattik There also a trigger which is for when a PR gets merged https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges-1.

I have never used it but just sharing it here.

**Comment by ouphi** [comment]
There is no corresponding story.

**Comment by algattik** [comment]
> There is no corresponding story.

correct

**Comment by algattik** [comment]
> @algattik There also a trigger which is for when a PR gets merged https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges-1.
> 
> I have never used it but just sharing it here.

Thanks, interesting. Still, it makes sense for downstream projects to run the pipelines when they refresh main, so they can ensure they have a working base setup

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (algattik) [reply .github/workflows/cd.yaml:4]: Thanks, implemented

**Review by cpeeyush — APPROVED** [review]

**Review by ouphi — APPROVED** [review]

**Pull request merged**